### PR TITLE
ADD - Created PHPUnit CI workflow

### PR DESCRIPTION
**Pull request by VinceAAU**
Created the PHPUnit GitHub Actions CI pipeline workflow file.

The reason that it's failing now is that the tests themselves are failing, which should be fixed by #31

### CODE REVIEW

**Review by ChristofferGamel — COMMENTED** [review]
er det meningen at phpunit tests skal fejle?

**Comment by DeeKahy** [comment]
@VinceAAU so the phpunit thingy got merged but this one is still failing

File: .github/workflows/phpunit.yml [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+            sleep 10 # Increased wait time to ensure MariaDB is ready
+
+            # Check if MariaDB socket file is created
+            if [ ! -S ${mariadb_socket} ]; then
+              echo "Error: MariaDB socket file not found at ${mariadb_socket}. Exiting..."
+              exit 1
+            fi
+          }
+
+          # I do not install adminer because I'm lazy
```
Comment (EmilChri) [inline .github/workflows/phpunit.yml:195]: why this comment?

Comment (ChristofferGamel) [inline .github/workflows/phpunit.yml:195]: ^^

Comment (VinceAAU) [reply .github/workflows/phpunit.yml]: The comment has been fixed to be more explanatory

Comment (VinceAAU) [reply .github/workflows/phpunit.yml]: The comment has been fixed to be more explanatory

**Review by EmilChri — APPROVED** [review]

File: .github/workflows/phpunit.yml [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+          -e '$a\\post_max_size=50M'
+          -e '$a\\upload_max_filesize=50M'
+          -e '$a\\extension=phar.so'
+          server/php/php.ini
+
+      # - name: Install composer
+      #   run: |
+      #     cd server/moodle
+      #     php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');"
+      #     php composer-setup.php
```
[the remark this answers is not in the document]
Comment (VinceAAU) [reply .github/workflows/phpunit.yml:51]: The `#` character means that the rest of the line is commented out. 
It is an alternative way to install composer, which we decided not to use in the end.

File: .github/workflows/phpunit.yml [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+          more php.ini
+          realpath php.ini
+
+      # - name: Copy php.ini to PHP-CLI directory
+      #   run: |
+      #     sudo cp server/php/php.ini /etc/php/8.2/cli/php.ini
+
+      # - uses: "ramsey/composer-install@v3"
+      #   with:
+      #     working-directory: "/etc/php/8.2/cli/php.ini"
```
[the remark this answers is not in the document]
Comment (VinceAAU) [reply .github/workflows/phpunit.yml:67]: The `#` character means that the rest of the line is commented out. 
It is an alternative way to install composer, which we decided not to use in the end.

File: .github/workflows/phpunit.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (ChristofferGamel) [reply .github/workflows/phpunit.yml:51]: Is it necessary to leave in then?

[the remark this answers is not in the document]
Comment (VinceAAU) [reply .github/workflows/phpunit.yml:51]: Not strictly, no. But this pull request focused on getting the CI pipeline up and running, and the cleanliness of the code was not a priority. Our team is currently in the process of optimising the workflow, and as part of this, the code will also be cleaned up. These changes will be introduced in a separate pull request.

**Review by ChristofferGamel — APPROVED** [review]

**Review by ZeroConfidence — APPROVED** [review]

**Pull request merged**